autoprecompiles/src/basic_blocks.rs

```diff
@@ -21,19 +23,23 @@ impl<T> SymbolicBlock<T> {
                 .iter()
                 .enumerate()
                 .map(|(i, instr)| format!("   instr {i:>3}:   {}", instr_formatter(instr)))
-                .collect::<Vec<_>>()
-                .join("\n")
+                .format("\n")
+                .to_string()
             + "\n])"
     }
 }
 
-pub struct SymbolicProgram<T> {
+/// Represents a symbolic program, which is a sequence of symbolic instructions
```

Comment:
A bit pedantic, but I think `symbolic` should also be removed here. The instructions and thus program are concrete.